### PR TITLE
Add replace_inf_values util function

### LINKED ISSUE #1496: Handle divide by 0 in DivideNumeric primitives
- The Binary Transform Primitive in Featuretools which will divide 2 numeric columns
- However, this primitives does not currently handle divide by 0 (it outputs **inf**, when it should output **NaN**)
- The DivideNumeric primitives should handle the divide by 0 case and output NaN, not `inf` like it currently does.
  - Ideally this happens without significantly affecting primitive performance speed 

### Open Questions
- Does this apply to DivideNumericScalar and DivideByFeature too?

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
### Add replace_inf_values util function
Closes #1496 

Adds a new utility function, `replace_inf_values`, that can be used to replace `np.inf` values in a feature matrix with `np.NaN` values (default) or with a value of the user's choosing (optional).

Also relocates Feature Selection section in API Reference, which was previously includes under the `EntitSet, Entity, Relationship, Variable Types` section instead of with the other feature related methods.

### CODE REVIEW

File: featuretools/computational_backends/utils.py [hunk 1]
```diff
@@ -274,3 +275,17 @@ def _check_cutoff_time_type(cutoff_time, es_time_type):
...
+
+        Args:
+            feature_matrix (DataFrame): DataFrame whose columns are feature names and rows are instances
+            replacement_value (int, float, str, optional): Value with which ``np.inf`` values will be replaced
+
+        Returns:
+            feature_matrix
+
+    """
+    return feature_matrix.replace([np.inf, -np.inf], replacement_value)
```
Comment (gsheni) [inline featuretools/computational_backends/utils.py:291]: Would it be using `inplace`?

Comment (thehomebrewnerd) [reply featuretools/computational_backends/utils.py:291]: We have generally avoided `inplace` operations as those are not supported by Dask.

Comment (gsheni) [reply featuretools/computational_backends/utils.py:291]: okay

File: docs/source/api_reference.rst [hunk 2]
```diff
@@ -251,6 +251,26 @@ Feature encoding
...
+.. currentmodule:: featuretools.selection
+.. autosummary::
+    :toctree: generated/
+
+    remove_low_information_features
+    remove_highly_correlated_features
+    remove_highly_null_features
+    remove_single_value_features
+
+Feature utils
```
Comment (rwedge) [inline docs/source/api_reference.rst:265]: how about Feature Matrix utils?

**Review by gsheni — APPROVED** [review]

**Review by rwedge — APPROVED** [review]

**Pull request merged**